index.js

```diff
@@ -89,22 +90,22 @@ exports.execute = async (srcName, query, params = {}, options = {}) => {
 
     console.debug(
       `Oracle Adapter: Connection secured: ${process.hrtime(start)[0]}s ${
-        process.hrtime(start)[1] / 1000000
+      process.hrtime(start)[1] / 1000000
       }ms`
     );
-    const result = await conn.execute(query, params, options);
+    result = await conn.execute(query, params, options);
 
     console.debug(
       `Oracle Adapter: Query executed: ${process.hrtime(start)[0]}s ${
-        process.hrtime(start)[1] / 1000000
+      process.hrtime(start)[1] / 1000000
       }ms`
     );
-
-    await conn.close();
-    return result;
   } catch (err) {
     console.error("Oracle Adapter: Error while executing query", err);
     throw new Error(err.message);
+  } finally {
+    await conn.close();
```

Comment:
conn is not available in this scope, since it is block scoped to the `try`.  We'll need to move this out as `let conn` similar to how you handled the result variable.

Reply:
I'll update

Reply:
I'll update that. Good catch!